172.md

```diff
@@ -0,0 +1,100 @@
+NIP-172
+=======
+
+Moderated Communities (Reddit Style)
+------------------------------------
+
+`draft` `optional` `author:vitorpamplona` `author:arthurfranca`
+
+The goal of this NIP is to create moderator-approved public communities around a topic. It defines the replaceable event `kind:34550` to define the community and the current list of moderators/administrators. Users that want to post into the community, simply tag any Nostr event with the community's `a` tag (See [NIP-33](33.md)). Moderators issue an approval event `kind:4550` that links the community with the new post.
+
+# Community Definition
+
+`Kind:34550` SHOULD include any field that helps define the community and the set of moderators.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": 34550,
+  "tags": [
+    ["d", "<Community name>"],
+    ["description", "<Community description>"],
+    ["image", "<Community image url>", "<Width>x<Height>"],
+	
+    //.. other tags relevant to defining the community
+
+    // moderators
+    ["p", "<32-bytes hex of a pubkey1>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey2>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey3>", "<optional recommended relay URL>", "moderator"],
+	
+    // relays used by the community 
+    ["relay", "<relay hosting author kind 0>", "author"],
+    ["relay", "<relay where to post requests to and fetch approvals from>"], 
+    ["relay", "<relay where to post requests to and fetch approvals from>"]
```

Comment:
```suggestion
    ["relay", "<relay where to post requests to>", "read"], 
    ["relay", "<relay where to fetch approvals from>", "write"],
    ["relay", "<relay where to post requests to and fetch approvals from>"]
```
Regarding relays to post requests to and fetch approvals from (not "author" mark), I was thinking it would be good to have optional "write" and "read" marks just like NIP-65 (and no mark meaning both read and write).

For example, it may be useful to focus reading post requests just from one "read" relay while the moderators can "write" the approvals to many relays.

Reply:
@vitorpamplona what do you think?

Reply:
`read/write` is very confusing because it depends on the point of view (moderator in this case). I wish we had better nomenclature to make it crystal clear. 

What about

```js
    ["relay", "<relay where to send and receive requests>", "requests"], 
    ["relay", "<relay where to send and receive approvals>", "approvals"],
    ["relay", "<relay where to post requests to and fetch approvals from>"]
```

It's not consistent with NIP65, but I think we need to start moving away from `read`/`write`

Reply:
Even NIP 65 isn't consistent with @mikedilger's model. He has something like 6 concepts for where things should be written to/from. This should eventually be added to nip 65.

Reply:
I agree `requests/approvals` is better.

Off-topic: @staab @mikedilger got me curious to know what are the new relay marks

Reply:
[danieletonon](https://coracle.social/npub10000003zmk89narqpczy4ff6rnuht2wu05na7kpnh3mak7z2tqzsv8vwqk) made a neat [infographic](https://coracle.social/note1emmc9zamttnf30r7mkjdenknl90llgu5hd00vyhzw7r7fgyn5pls0azas4) with sort of an "inbox model". I'm not sure how this would be formalized.

Reply:
Markers added.